src/app/pages/TopicView/TopicView.tsx

```diff
@@ -55,24 +63,46 @@ function TopicView(): JSX.Element {
               <DoubleChevronLeft width="24" /> <h2> {title}</h2>
             </TitleContainer>
             <Description>{description}</Description>
-            <h3>Needs</h3>
-            {needs.length > 0 ? (
-              <NeedsList>
-                {needs.map(need => (
-                  <Need
-                    key={need.id}
-                    content={need}
-                    onUpvoteChange={handleNeedUpvote(need.id)}
-                  />
-                ))}
-              </NeedsList>
+            <SliderMenu
+              options={menuCategories}
+              selectedOption={category}
+              onSelect={category => setCategory(category)}
+            />
+            {category === 'needs' ? (
+              <>
+                {needs.length > 0 ? (
+                  <NeedsList>
+                    {needs.map(need => (
+                      <Need
+                        key={need.id}
+                        content={need}
+                        onUpvoteChange={handleNeedUpvote(need.id)}
+                      />
+                    ))}
+                  </NeedsList>
+                ) : (
+                  <Disclaimer>no needs added yet.</Disclaimer>
+                )}
+                <Button onClick={() => setView('SHOW_NEED_FORM')}>
+                  Add Need
+                </Button>
+              </>
             ) : (
-              <Disclaimer>no needs added yet.</Disclaimer>
+              <>
+                {proposals.length > 0 ? (
+                  <ProposalList>
+                    {proposals.map(proposal => (
+                      <Proposal content={proposal} key={proposal.id} />
+                    ))}
+                  </ProposalList>
+                ) : (
+                  <Disclaimer>no proposals added yet.</Disclaimer>
+                )}
+                <Button highlight onClick={() => nav('addProposal')}>
+                  Add Proposal
+                </Button>
+              </>
             )}
```

Comment:
This seems a little nested and therefore hard to read. To "clean up" a little, you could for example create an if-condition for the content of your TopicView and return it as `{content}`.

Here a short example of what I mean: 

```
export default function TopicView() {

let content; 

if (category === 'needs') {
 content = ( 
  <NeedsList>
     {needs.map(need => (
      <Need
         key={need.id}
         content={need}
         onUpvoteChange={handleNeedUpvote(need.id)}
        />
   ))}
 </NeedsList>
);
} else {
 content = (
    <ProposalList>
       {proposals.map(proposal => (
        <Proposal 
           content={proposal} 
           key={proposal.id} />
        ))}
     </ProposalList>
);
}

return (
<div>{content}</div>
)
}
```